BrowserKit/Sources/MenuKit/MenuDetailView.swift

```diff
@@ -0,0 +1,57 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import UIKit
+
+public final class MenuDetailView: UIView,
+                                   MenuTableViewNavigationDelegate {
+    // MARK: - UI Elements
+    private var tableView: MenuTableView = .build()
+    private var detailHeaderView: MenuSubmenuHeaderView = .build()
+
+    // MARK: - Initializers
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - UI Setup
+    private func setupView() {
+        backgroundColor = .clear
+        self.addSubview(detailHeaderView)
+        self.addSubview(tableView)
+
+        NSLayoutConstraint.activate([
+            detailHeaderView.topAnchor.constraint(equalTo: self.topAnchor),
+            detailHeaderView.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            detailHeaderView.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+            detailHeaderView.heightAnchor.constraint(equalToConstant: 70),
+
+            tableView.topAnchor.constraint(equalTo: detailHeaderView.bottomAnchor),
+            tableView.bottomAnchor.constraint(equalTo: self.bottomAnchor),
+            tableView.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            tableView.trailingAnchor.constraint(equalTo: self.trailingAnchor)
+        ])
+    }
+
+    // MARK: - Interface
+    public func setupTableViewNavigationDelegate(with delegate: any MenuTableViewNavigationDelegate) {
+        tableView.navigationDelegate = delegate
```

Comment:
Would we be able to dispatch an action here instead of creating a delegate? 

Could it be possible to dispatch an action say `showSettings`, then the view will receive the new state and  which can check `state.showSettings == true` in `newState` and then call the appropriate method whether it belongs to the VC or coordinator? Not sure if I'm interpreting this correctly, but that's my first thought looking at this.

Reply:
We spoke about this on a chat and it will be done when I'm adding redux to the detailVC in the next PR. :)